lib/compat/wordpress-6.7/rest-api.php

```diff
@@ -114,3 +114,45 @@ function gutenberg_override_default_rest_server() {
 	return 'Gutenberg_REST_Server';
 }
 add_filter( 'wp_rest_server_class', 'gutenberg_override_default_rest_server', 1 );
+
+/**
+ *
+ * This function is hooked to the `rest_prepare_comment` filter and is responsible for updating the comment metadata based on the data provided in the REST API request.
+ *
+ * It performs the following tasks:
+ * - Updates the `block_comment` metadata for the comment based on the `meta` field in the request.
+ * - Updates the `comment_type` and `comment_approved` fields for the comment based on the corresponding fields in the request.
+ * - Retrieves the author's avatar URLs and adds them to the response data.
+ * - Retrieves the `block_comment` metadata for the comment and adds it to the response data.
+ *
+ * @param WP_REST_Response $response The response object.
+ * @param WP_Comment       $comment  The comment object.
+ * @param WP_REST_Request  $request  The request object.
+ * @return WP_REST_Response The updated response object.
+ */
+if ( ! function_exists( 'update_comment_meta_from_rest_request_6_7' ) ) {
+	function update_comment_meta_from_rest_request_6_7( $response, $comment, $request ) {
+
+		if ( ! empty( $request['comment_type'] ) ) {
+			$comment_data = array(
+				'comment_ID'       => $comment->comment_ID,
+				'comment_type'     => $request['comment_type'],
+				'comment_approved' => isset( $request['comment_approved'] ) ? $request['comment_approved'] : 0,
+			);
+
+			wp_update_comment( $comment_data );
+		}
+
+		if ( $response->data['author'] ) {
+			$avatar_url                           = get_avatar_url( $response->data['author'] );
+			$response->data['author_avatar_urls'] = array(
+				'default' => $avatar_url,
+				'48'      => add_query_arg( 's', 48, $avatar_url ),
+				'96'      => add_query_arg( 's', 96, $avatar_url ),
+			);
+		}
+
+		return $response;
+	}
+}
+add_filter( 'rest_prepare_comment', 'update_comment_meta_from_rest_request_6_7', 10, 3 );
```

Comment:
Using the `get_comment` filter sets the avatar URL to null, so we've switched to using the `rest_prepare_comment` filter instead. This ensures the avatar URL is properly included in the comment response.

Reply:
I understand. But this will only work when retrieving comments via the REST API. If we want to retrieve them on the backend, it won't work. 

I'm trying to understand why `get_comment` will set the avatar URL to `null`. A few possibilities come to mind:

- is it because the `show_avatars` option is not enabled? 
- is it because the `author_avatar_urls` field is not enabled when performing the REST API request?

Can you speak more about where this assignment to `null` happens?

In any case, to achieve what you're doing it should be enough to just use the `rest_avatar_sizes` filter and the `rest_get_avatar_urls()` function would build the avatar URLs properly. If we're just doing it for the REST API response, we're doing it in an inconsistent way which is likely to introduce unexpected bugs in unexpected places.